Adds the Planck EZ, 3737 RGB, fixes out-of-tune notes

jackhumbert opened the pull request:
Adds the Planck EZ, RGB settings for the IS31FL3737 chip, and fixes some issues I was noticing with out-of-tune notes with the Arm audio setup. This can be rebased on #5372 once that's merged.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [x] Core
- [x] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

drashna commented on keyboards/planck/ez/config.h:
```diff
@@ -0,0 +1,144 @@
+//#define FORCE_NKRO
+
+/* key combination for magic key command */
+#define IS_COMMAND() ( \
```
This isn't needed, since it's the default.

drashna commented on keyboards/planck/ez/config.h:
```diff
@@ -0,0 +1,144 @@
+
+#define MUSIC_MAP
+#undef AUDIO_VOICES
+#undef C6_AUDIO
```
No audio?

jackhumbert replied:
IIRC, this doesn't do anything in Arm right now :/

drashna replied:
Oh, ... yeah, then wouldn't be needed. :D

drashna commented on quantum/rgb_matrix_drivers.c:
```diff
@@ -33,23 +33,29 @@ static void init( void )
     IS31FL3731_init( DRIVER_ADDR_1 );
     IS31FL3731_init( DRIVER_ADDR_2 );
-#else
+#elif defined(IS31FL37313)
```
I think you meant: 
```suggestion
#elif defined(IS31FL3733)
```

drashna commented:
Default keymap compiles, at least.

yanfali commented on keyboards/planck/ez/ez.c:
```diff
@@ -0,0 +1,175 @@
+      |             {x=0..224, y=0..64}
+      |              |                 modifier
+      |              |                 | */
+    {{0|(0<<4)},   {20.36*0, 21.33*0}, 1},
```
I feel like this should just be a define and re-used across all the rows except the last one.

yanfali commented on quantum/stm32/mcuconf.h:
```diff
@@ -154,7 +154,11 @@
  * I2C driver system settings.
  */
-#define STM32_I2C_USE_I2C1                  FALSE
+// #ifdef STM32_I2C
```
can you just delete these commented out lines?

jackhumbert commented:
> No RGB_MATRIX_KEYPRESSES? =(

Some other things need to be addressed with this. The matrix for the rev 6/ez has the right side below the left, so it doesn't work out-of-box :/

XScorpion2 commented:
> 
> 
> > No RGB_MATRIX_KEYPRESSES? =(
> 
> Some other things need to be addressed with this. The matrix for the rev 6/ez has the right side below the left, so it doesn't work out-of-box :/
> 
> edit: I'm not 100% sure I updated this from the Planck Light's config.. checking that now.

Interesting, I reviewed the rgb_led initialization and it looked like it should work out of the box (but slow until the overhaul pr lands). Actually, now that you mention it I also didn't look too close at the matrix_co field and it indeed is incorrect...so we both missed it it seems.

jackhumbert commented:
> > > No RGB_MATRIX_KEYPRESSES? =(
> > 
> > 
> > Some other things need to be addressed with this. The matrix for the rev 6/ez has the right side below the left, so it doesn't work out-of-box :/
> > edit: I'm not 100% sure I updated this from the Planck Light's config.. checking that now.
> 
> Interesting, I reviewed the rgb_led initialization and it looked like it should work out of the box (but slow until the overhaul pr lands). Actually, now that you mention it I also didn't look too close at the matrix_co field and it indeed is incorrect...so we both missed it it seems.

I corrected the `g_rgb_leds` config and things work perfectly now :)

jackhumbert commented:
Pushed to `planck_ez_with_5372` - will merge that into this branch once #5372 is merged :) (now approved)

XScorpion2 commented:
> 
> 
> Pushed to `planck_ez_with_5372` - will merge that into this branch once #5372 is merged :) (now approved)

@drashna You heard the man! MERGE IT! =D

moyi4681 commented:
is it possilbe to support two is31fl3737, like is31fl3731,  so we can use 96 rgb leds:)

XScorpion2 commented:
At that point I would just switch to ws2812 LEDs as then you can easily extend the strand length to whatever you want. https://github.com/qmk/qmk_firmware/pull/5467

drashna commented:
This looks like it's ready to go?

@moyi4681 I think so?  
@XScorpion2 why is that?  I suspect that the power draw would be greater with the WS2812 strip, than with the matrix (but I could be wrong).  Like more than 30 LEDs on the WS2812 can cause power issues.

drashna commented:
@jackhumbert Could you update the RGB Matrix doc so it includes reference to the 3737 driver? 
Also, does this driver support dual controllers, as is?

XScorpion2 commented:
@drashna single pin for data, easy to extend to any length. I think there are variations of the ws2812 that take less power. But that's a real concern no matter which approach you use... How many rgb LEDs can I cram into this kb before I burn my house down. So far I'm up to 152.

moyi4681 commented:
@XScorpion2 there are 3 reason I don't want to use ws2812 for rgb matrix:
1. it's hard to use ws2812 for SMT, usually we need make 2 pcbs together;
2. if one ws2812 broken, it will cost a BIG problem: rest of them who connect to the broken one can't work anymore; 
3. it's more expensive than normal rgb LEDS,

drashna commented:
> * it's hard to use ws2812 for SMT, usually we need make 2 pcbs together;
> * if one ws2812 broken, it will cause a BIG problem: rest of them who connect to the broken one can't work anymore;

OMG. As an owner of the Corne Keyboard (helidox), I can attest to just how much of a nightmare this can be!

drashna commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Planck EZ](https://trello.com/c/iSx0STUy/8-planck-ez)

noroadsleft commented on docs/feature_rgb_matrix.md:
```diff
@@ -52,7 +52,9 @@ const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
-###  IS31FL3733
+###  IS31FL3733/IS31FL3737
+
+?> For the IS31FL3737, replace all instances of `IS31FL3733` below with `IS31FL3737`
```
IMO, this should be tagged as an Important Notice instead of a General Tip.

```suggestion
!> For the IS31FL3737, replace all instances of `IS31FL3733` below with `IS31FL3737`.
```

jackhumbert replied:
Thanks!
